src/profile-logic/profile-data.js

```diff
@@ -161,6 +165,108 @@ export function getSampleCallNodes(
   });
 }
 
+export type SelectedState =
+  // Samples can be filtered through various operations, like searching, or
+  // call tree transforms.
+  | 'FILTERED_OUT'
+  // This sample is selected because either the tip or an ancestor call node matches
+  // the currently selected call node.
+  | 'SELECTED'
+  // This call node is not selected, and the stacks come before the selected call node.
+  | 'BEFORE_SELECTED'
+  // This call node is not selected, and the stacks come after the selected call node.
+  | 'AFTER_SELECTED';
+
+/**
+ * Go through the samples, and determine their current state. See the comments on the
+ * the SelectedState type definition for more information.
```

Comment:
Let's add some more explanation to this comment.

```
 * For samples that are neither 'FILTERED_OUT' nor 'SELECTED', this function compares
 * the sample's call node to the selected call node, in tree order. It uses the same
 * ordering as the function compareCallNodes in getTreeOrderComparator. But it does not
 * call compareCallNodes with the selected node for each sample's call node, because doing
 * so would recompute information about the selected call node on every call. Instead, it
 * has an equivalent implementation that is faster because it only computes information
 * about the selected call node's ancestors once.
```